lib/FunctionIsolation/EnforceABI.cpp

```diff
@@ -91,6 +92,29 @@ bool EnforceABI::runOnModule(Module &M) {
   return false;
 }
 
+static std::optional<model::RawFunctionType>
+makeRawPrototypeOrDefault(const model::Type *Type, model::Binary &TheBinary) {
```

Comment:
Given that we said that the downgrading might create new structs, this API won't do, I'm afraid.

I'm tempted to leave out these kind of changes for now, and find a solution later on.

Reply:
This function is also an attempt to speed the PR up. There was something similar in your initial implementation - I just moved it closer to the only user. And yes, I like this no more than you do.

Reply:
So, after the other day's discussion I think we no longer need to create new `struct`s, so this is fine! Marking as solved.